### PR TITLE
[Reflection] CoreFX import for Module type

### PR DESCRIPTION
**Pull request by MaximLipnin**
Part of #9660.

The changes:
- included corefx version of Module type;
- enabled related xunit tests from corefx System.Reflection and System.Runtime locations;
- enabled PointerTests fixture which some Module tests depend on;
- made Mono implemented Module class partial; removed duplicated code and left only Mono internal stuff;
- updated corlib make-file to process TestModule.il required by some Module tests (but they will be disabled due to runtime crash).

### CODE REVIEW

**Comment by lambdageek** [comment]
I wonder if we should move the internal calls, such as
https://github.com/mono/mono/blob/22143c7916d8f0b2095aeed4d1c4bf0c8cbfedc6/mcs/class/corlib/System.Reflection/Module.cs#L238-L257
from Mono's `System.Reflection/Module.cs` to `MonoModule.cs`

Maybe we should make that a cleanup task for the future.

**Comment by lambdageek** [comment]
Or perhaps, given the build errors, it ought to happen now.  And perhaps the private fields that are accessed from the runtime ought to be on MonoModule.  I think all concrete instances are `MonoModule`s?

**Comment by vargaz** [comment]
Will look into that.

**Comment by MaximLipnin** [comment]
https://github.com/mono/mono/pull/11595 needs to be merged to fix a bug with parsing the nunit xml from xunit.

**Comment by akoeplinger** [comment]
@monojenkins build

**Comment by MaximLipnin** [comment]
@monojenkins build failed

File: mcs/class/corlib/Makefile [hunk 1]
```diff
@@ -133,8 +133,13 @@ DEFAULT_REFERENCES =
 
 TEST_LIB_REFS = System.Core System
 
+TestModule.dll: ../../../external/corefx/src/System.Runtime/tests/TestModule/System.Reflection.TestModule.il
```
Comment (marek-safar) [inline mcs/class/corlib/Makefile]: This need to use profile specific path, e.g. `$(test_lib_dir)/System.Reflection.TestModule.dll:`

File: mcs/class/corlib/System.Reflection/Module.cs [hunk 2]
```diff
@@ -246,199 +134,18 @@ internal virtual Guid GetModuleVersionId ()
...
-			get { return GetCustomAttributesData (); }
-		}
-
 		internal virtual IntPtr GetImpl () {
 			throw CreateNIE ();
 		}
+
+		static Exception CreateNIE ()
+		{
+			return new NotImplementedException ("Derived classes must implement it");
```
Comment (marek-safar) [inline mcs/class/corlib/System.Reflection/Module.cs]: Use existing `throw NotImplemented.ByDesign;`

File: mcs/class/corlib/System.Reflection/Module.cs [hunk 3]
```diff
@@ -43,97 +43,17 @@ internal enum ResolveTokenError {
...
-		Module () {
-		}
-
-		public ModuleHandle ModuleHandle {
-			get {
-				return new ModuleHandle (GetImpl ());
-			}
-		}
-
+	public abstract partial class Module : ISerializable, ICustomAttributeProvider {
```
Comment (marek-safar) [inline mcs/class/corlib/System.Reflection/Module.cs]: Could be just simple `partial class Module` there rest is copied from corefx part

**Comment by marek-safar** [comment]
The change is ok but there is a strangely consistent mono_threads_suspend_begin_async_suspend issue on macOS only. @lambdageek @luhenry could you check what this is about?

**Comment by MaximLipnin** [comment]
Does it have a chance to be merged?

**Comment by luhenry** [comment]
@monojenkins build failed

**Comment by lambdageek** [comment]
I'm going to spend a couple of hours to try to repro the osx suspend issue - I don't think I've seen this kind of failure before.

**Comment by lambdageek** [comment]
Tried to get osx64 to fail locally a half-dozen times or so.  No luck.  Looks like it passed the most recent run on CI, too.

**Comment by marek-safar** [comment]
@monojenkins commit csproj and apidiff

**Comment by MaximLipnin** [comment]
Thanks for review!

**Pull request merged**